src/ui/listbox.cpp

```diff
@@ -249,9 +249,11 @@ bool ListBox::onProcessMessage(Message* msg)
           if (dynamic_cast<ui::Separator*>(picked))
             picked = nullptr;
 
-          // If the picked widget is a child of the list, select it
-          if (picked && hasChild(picked))
-            selectChild(picked, msg);
+          // If the picked widget is a descendant of the list, select the item containing it.
+          if (picked && hasDescendant(picked)) {
```

Comment:
Probably this can use the `hasAnscestor()` (which doesn't have to iterate through all children of children, and it's already used in other places):
```suggestion
          if (picked && picked->hasAncestor(this)) {
```